### DISCUSSION TITLE
Can i we self hosted windows runners as a specific user and not Administrator by default ?

### DISCUSSION THREAD

**Discussion by msajidmansoori12** [community/community#139870] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

I'm trying to provision windows runner dynamically on aws , our use case needs us to run the scripts as a non administrative user but i'm not sure if that's something doable or we can't do that. 
Any suggestions , tips would be appreciated.

**Comment by lovrozagar** [comment]
Hi @msajidmansoori12 ,

Provisioning a Windows runner dynamically on AWS to run scripts as a non-administrative user is possible! Here are some tips to help you get started:

IAM Roles: Make sure to configure the appropriate IAM roles and policies in AWS that grant your runner the necessary permissions while keeping it limited to non-administrative actions.

User Creation: When you set up your Windows instance, create a non-administrative user during the provisioning process. You can use the EC2 user data feature to run a script at instance startup to create this user and configure it as needed.

Remote Desktop Protocol (RDP): If you need to interact with the instance, ensure that your non-admin user has the required permissions for RDP access, but avoid giving admin rights to this user.

Automation Scripts: Use tools like AWS CloudFormation or Terraform to automate the provisioning process. You can include steps to set up your non-administrative user in your scripts.

Testing: Once your runner is provisioned, test running scripts with the non-admin user to ensure that everything works as expected. You may need to adjust permissions based on the scripts you intend to run.

If you run into specific issues or have questions about any of these steps, feel free to ask! Good luck with your setup!
